### DISCUSSION TITLE
how to use Cesiumjs function for realitydata in itwin 3.0 or imodeljs 2.*

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#3273] [Q&A]
how to use Cesiumjs function for realitydata in itwin 3.0 or imodeljs 2.* ??
for example  i want change the realitydata  transform in  Cesiumjs 

 let newtr = update3dtilesMaxtrix(px, py, gd, scale);

   tileset.readyPromise.then(function (tileset) {
        tileset._root.transform = newtransform;
        if (zoom) {

            viewer.zoomTo(tileset);
        }
    });

**Comment by pmconne** [comment]
What is the type of `viewer` - is it a Cesium viewer, or an iTwin viewer?
Trying to use both Cesium and iTwin.js in the same application sounds less than ideal. Are you aware that iTwin.js supports the same 3d tilesets that Cesium can?

**Reply by paprophet** [reply]
the code view is cesium viewer  because the code in  Cesiumjs  ,i know iTwin.js supports the same 3d tilesets that Cesium can
so i want know how can  iTwin.js support function same the  Cesium can,for example change the transform.

**Reply by pmconne** [reply]
Thanks. We can add an API for applying a transform to a reality model. Are there other functions you need and if so can you list them please?

**Reply by paprophet** [reply]
thank you , 
1. Cesium view can add entity for example:
viewer.scene.primitives.add(
            new Cesium.ClassificationPrimitive({
                geometryInstances: new Cesium.GeometryInstance({
                    geometry: Cesium.PolygonGeometry.createGeometry(polygonGeometry),
                    attributes: {
                        color: Cesium.ColorGeometryInstanceAttribute.fromColor(
                            Cesium.Color.fromRandom({ alpha: 0.8 })
                        ),
                        show: new Cesium.ShowGeometryInstanceAttribute(true),
                    },
                    id: id 
                }),
                classificationType: Cesium.ClassificationType.CESIUM_3D_TILE,
            }) 
2.if the itwin can import and support cesiumjs,and user can use the cesium function direct to analysis realitydata the itwin viewer  is perfect ,because the cesium have many methods to analysis realitydata
